src/gui/widgets/color_wheel_widget.cpp

```diff
@@ -0,0 +1,435 @@
+	              p.x(), p.y());
+}
+
+/// Rotation of the selector image
+qreal ColorWheel::selectorImageAngle() const
+{
+	return -hue * 360 - 60;
+}
+
+/// Offset of the selector image
+QPointF ColorWheel::selectorImageOffset()
+{
+	return QPointF(-innerRadius(), -triangleSide() / 2);
+}
+
+/// Calculate the side of the inner triangle
+qreal ColorWheel::triangleSide() const
+{
+	return innerRadius() * qSqrt(3);
+}
+
+/// Calculate the height of the inner triangle
+qreal ColorWheel::triangleHeight() const
+{
+	return innerRadius() * 3 / 2;
+}
+
+
+ColorPreview::ColorPreview(QWidget* parent)
+    : QFrame(parent)
+{
+	setFrameStyle(QFrame::Panel | QFrame::Plain);
+	setLineWidth(2);
+}
+
+
+/// Get widget face color
+QColor ColorPreview::color()
+{
+	return display_color;
+}
+
+
+/// Set widget face color
+void ColorPreview::setColor(QColor c)
+{
+	display_color = c;
+	update();
+}
+
+
+void ColorPreview::paintEvent(QPaintEvent* event)
+{
+	QFrame::paintEvent(event);
+
+	QPainter painter(this);
+	painter.fillRect(contentsRect(), QBrush(display_color));
+}
+
+
+QSize ColorPreview::sizeHint() const
+{
+	return QSize(24, 24);
+}
+
+
+ColorWheelDialog::ColorWheelDialog(QWidget* parent, Qt::WindowFlags f)
+    : QDialog(parent, f)
+{
+	auto* main_layout = new QVBoxLayout();
+	wheel = new ColorWheel();
+	main_layout->addWidget(wheel);
+	auto* mid_layout = new QHBoxLayout();
+	auto* color_preview = new ColorPreview();
+	color_preview->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
+	mid_layout->addWidget(color_preview);
+	text_edit = new QLineEdit();
+	text_edit->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Fixed);
+	mid_layout->addWidget(text_edit);
+	main_layout->addLayout(mid_layout);
+	auto* ok_button = new QPushButton(tr("Select"));
+	ok_button->setDefault(true);
+	main_layout->addWidget(ok_button);
+	setLayout(main_layout);
+
+	connect(wheel, &ColorWheel::colorSelected, this, [this](QColor c) {
+		text_edit->setText(c.name().right(6).toUpper());
+	});
+	connect(text_edit, &QLineEdit::textEdited, this,
+	        [this](const QString& text) {
+		        const auto c = QRgb(text.toUInt(nullptr, 16));
+				wheel->setColor(c);
+	        });
+	connect(wheel, &ColorWheel::colorChanged, color_preview,
+	        &ColorPreview::setColor);
+	connect(ok_button, &QPushButton::clicked, this, &ColorWheelDialog::accept);
+}
+
+
+QColor ColorWheelDialog::currentColor() const
+{
+	return result() == QDialog::Accepted ? wheel->color() : QColor();
+}
+
+
+void ColorWheelDialog::setColor(const QColor& c)
+{
+	wheel->setColor(c);
+	text_edit->setText(c.name().right(6).toUpper());
+	// color preview receives the update from the color wheel
+}
+
+
+QColor ColorWheelDialog::getColor(const QColor& initial, QWidget* parent)
+{
+	ColorWheelDialog dlg(parent);
+	dlg.setColor(initial);
+	auto const sizex = QGuiApplication::primaryScreen()->availableSize();
+	auto const dlg_size = qMin(sizex.width(), sizex.height());
+	dlg.resize(QSize(dlg_size, dlg_size));
```

Comment:
Is this for maximizing the dialog?
This is what I use in `TextBrowserDialog`:
https://github.com/OpenOrienteering/mapper/blob/d3fc644a8b6c4f5958704a315a7b252293ba071a/src/gui/text_browser_dialog.cpp#L87-L88

Reply:
Yes, it is. Your code is much better and I'll adopt it. Thanks for the suggestion!